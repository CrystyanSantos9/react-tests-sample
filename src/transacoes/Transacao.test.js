import React from "react";
import { render } from "@testing-library/react";
import Transacao from "./Transacao";

describe('Componente de criação das informações da lista de extrato',()=>{
    it('O snapshot do component deve permanecer sempre o mesmo', ()=>{
        //uso o elemento container do objeto render
        const {container} = render(<Transacao 
            data="08/10/2022"
            tipo="saque"
            valor="85.00"
        />)
        //para pegar o html criado pela função render referente ao meu componente
        //uso o método firstChild do objeto container de render
        expect(container.firstChild).toMatchSnapshot();

    })
})