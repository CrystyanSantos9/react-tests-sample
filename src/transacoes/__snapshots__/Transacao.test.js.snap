// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Componente de criação das informações da lista de extrato O snapshot do component deve permanecer sempre o mesmo 1`] = `
<div
  class="transacao-container"
>
  <p>
    08/10/2022
  </p>
  <p>
    saque
  </p>
  <p>
    R$ 
    85.00
  </p>
</div>
`;
